File: CommunityHub/CommunityHub.API/Models/Community.cs
namespace CommunityHub.API.Models;

public class Community : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
    public bool IsDeleted { get; set; } = false;

    public List<Event> Events { get; set; } = [];
}

File: frontend/src/models/Community.ts
import { Event } from './Event';

export interface Community {
  id: string;
  name: string;
  description: string;
  category: string;
  city: string;
  country: string;
  ownerId: string;
  events: Event[];
}

export interface CommunitySearchRequest {
  keywords?: string;
  location?: string;
  category?: string;
  page: number;
  pageSize: number;
}
